### PR TITLE
Migrate to Jakarta validation

### PR DESCRIPTION
**Pull request by dstepanov**
Removes all references to `javax.validation` moving to jakarta

### CODE REVIEW

File: inject/src/test/groovy/io/micronaut/context/env/yaml/YamlPropertySourceLoaderSpec.groovy [hunk 1]
```diff
@@ -46,7 +46,6 @@ class YamlPropertySourceLoaderSpec extends Specification {
             @Override
             protected SoftServiceLoader<PropertySourceLoader> readPropertySourceLoaders() {
                 GroovyClassLoader gcl = new GroovyClassLoader()
-                gcl.addClass(YamlPropertySourceLoader)
```
Comment (sdelamo) [inline inject/src/test/groovy/io/micronaut/context/env/yaml/YamlPropertySourceLoaderSpec.groovy:49]: why this removal?

File: jackson-core/src/test/groovy/io/micronaut/jackson/core/env/JsonPropertySourceLoaderSpec.groovy [hunk 2]
```diff
@@ -90,7 +91,6 @@ class JsonPropertySourceLoaderSpec extends Specification {
             @Override
             protected SoftServiceLoader<PropertySourceLoader> readPropertySourceLoaders() {
                 GroovyClassLoader gcl = new GroovyClassLoader()
-                gcl.addClass(JsonPropertySourceLoader)
```
Comment (sdelamo) [inline jackson-core/src/test/groovy/io/micronaut/jackson/core/env/JsonPropertySourceLoaderSpec.groovy:93]: why this removal?

**Review by sdelamo — APPROVED** [review]

File: inject/src/test/groovy/io/micronaut/context/env/yaml/YamlPropertySourceLoaderSpec.groovy (hunk 1 above)
Comment (dstepanov) [reply inject/src/test/groovy/io/micronaut/context/env/yaml/YamlPropertySourceLoaderSpec.groovy:49]: That solves Groovy problems

**Comment by nbrugger-tgm** [comment]
Linked issue : https://github.com/micronaut-projects/micronaut-core/issues/4770

**Pull request merged**